Changes to Fuel and PortionOfMatter hierarchies

jannahastings opened the pull request:
Remove fuel subclasses of 'energy carrier' disposition hierarchy. Rename energy carrier dispositions to add -disposition suffix. 

Add FuelRole beneath role. 

In PortionOfMatter hierarchy: 

- remove 'PortionOf' prefixes

- add CombustionFuel

- add appropriate axioms linking substances that can be fuels to the fuel role and the appropriate energy carrier disposition

Closes #227. 

When reviewing please check the *inferred* hierarchy after running classification with the reasoner.

akleinau commented:
the EnergyCarrierDisposition definition is not aristotelean yet: _"According to ISO 13600, an energy carrier is either a substance or a phenomenon that can be used to produce mechanical work or heat or to operate chemical or physical processes. ..."_ It's also rather long, can we shorten it?

The inferred fuel hierarchy is different to what PortionOfFuel had before, e.g. OilAndPetroleumProducts is no direct subclass of PortionOfFuel anymore, Biofuel is CombustionFuel and RenewableFuel, ... is that intended? 

can the "WoodAndOther" class be renamed?

akleinau commented:
I added some term tracker items :)

l-emele commented:
> the EnergyCarrierDisposition definition is not aristotelean yet: _"According to ISO 13600, an energy carrier is either a substance or a phenomenon that can be used to produce mechanical work or heat or to operate chemical or physical processes. ..."_ It's also rather long, can we shorten it?

I suggest we leave the definition in this pull request as it is and create a new issue to discuss this definition.

> The inferred fuel hierarchy is different to what PortionOfFuel had before, e.g. OilAndPetroleumProducts is no direct subclass of PortionOfFuel anymore, Biofuel is CombustionFuel and RenewableFuel, ... is that intended?

Yes, `Biofuel` is both a `CombustionFuel` and a `RenewableFuel`. But are there some renewable fuels that are no combustion fuels? I don't think so, hence `RenewableFuel` should be (at least in the inferred hierarchy) a child of `CombustionFuel`.

> can the "WoodAndOther" class be renamed?

Maybe simply `Wood`? The _and other part_ is still covered by the upper class `SolidBiomass`.

l-emele commented:
> > can the "WoodAndOther" class be renamed?
> 
> Maybe simply `Wood`? The _and other part_ is still covered by the upper class `SolidBiomass`.

Maybe it is better to keep it for the moment create also a separate issue.

stap-m commented:
We already discussed the EnergyCarrier def in #209: *An EnergyCarrier is a disposition of an object or objectAggregate that contains energy for conversion as usable energy.*
Instead of 'EnergyCarrier' it would be now 'EnergyCarrierDisposition' though. Do you still agree with the definition?

l-emele commented:
So we leave EnergyCarrier / EnergyCarrier disposition here as it is and fix it in #209?

stap-m commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -763,6 +955,17 @@ Genreally, flows in the model can be controlled/optimized."@en,
     
+Class: DammedWater
+
+    Annotations: 
```
The subclasses of water need some attention. The def of dammed water is on dams, not on dammed water!!! 
But this is off-topic here.

jannahastings replied:
Maybe open a different issue? :-)

l-emele replied:
I think we should leave water and its subclasses in this PR as it is and shift it to another issue. Either #43 or #78 or a completely new issue.

stap-m commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -925,7 +1140,7 @@ Class: Energy
+Class: EnergyCarrierDisposition
 
     Annotations: 
         <http://purl.obolibrary.org/obo/IAO_0000115> "According to ISO 13600, an energy carrier is either a substance or a phenomenon that can be used to produce mechanical work or heat or to operate chemical or physical processes. It is any system or substance that contains energy for conversion as usable energy later or somewhere else. This could be converted for use in, for example, an applian ...
```
```suggestion
        <http://purl.obolibrary.org/obo/IAO_0000115> "An EnergyCarrier is a disposition of an object or objectAggregate that contains energy for conversion as usable energy."@en,
```

stap-m commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -1911,629 +2371,108 @@ pull request: https://github.com/OpenEnergyPlatform/ontology/pull/234"
     Annotations: 
-        <http://purl.obolibrary.org/obo/IAO_0000115> "High rank coal used for industrial and residential applications. It has generally less than 10 % volatile matter and a high carbon content (about 90 % fixed carbon). Its gross calorific value is greater than 23 865 kJ/kg (5 700 kcal/kg) on an ash-free but moist basis."@en,
-        <http://purl.obolibrary.org/obo/IAO_0000119> "REGULATION (EC) No 1099/2008 OF THE EUROPEAN PARLIAMENT AND OF THE COUNCIL of 22 October 2008 on energy statistics https://eur-lex.europa.eu/legal-content/EN/TXT/HTML/?uri=CELEX:32008R1099&from=EN"@en
+        <http://purl.obolibrary.org/obo/IAO_0000115> "A portion of matter isa part of a material entity that has a state_of_matter property."@en,
```
```suggestion
        <http://purl.obolibrary.org/obo/IAO_0000115> "A portion of matter is a part of a material entity that has a state_of_matter property."@en,
```
Can we keep this definition and not refer to the state of matter in its subclasses?

stap-m commented:
> So we leave EnergyCarrier / EnergyCarrier disposition here as it is and fix it in #209?

Sorry, I didn't see your comment. I already made a change request for this here....

akleinau commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -522,6 +530,18 @@ Class: AgriculturalDemand
+Class: Air
+
+    Annotations: 
+        <http://purl.obolibrary.org/obo/IAO_0000115> "Air is a portion of a composition of gases that forms the Earth's atmosphere."@en,
```
```suggestion
        <http://purl.obolibrary.org/obo/IAO_0000115> "Air is a portion of matter that consists of a composition of gases that forms the Earth's atmosphere."@en,
```

akleinau commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -522,6 +530,18 @@ Class: AgriculturalDemand
+Class: Air
+
+    Annotations: 
+        <http://purl.obolibrary.org/obo/IAO_0000115> "Air is a portion of a composition of gases that forms the Earth's atmosphere."@en,
```
"consists of" maybe?

jannahastings commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -353,19 +342,19 @@ pull request: https://github.com/OpenEnergyPlatform/ontology/pull/234"
-        <http://purl.obolibrary.org/obo/RO_0000091> some CombustionFuel,
+        <http://purl.obolibrary.org/obo/RO_0000091> some CombustibleEnergyCarrierDisposition,
         <http://purl.obolibrary.org/obo/RO_0000091> some GreenhouseEffectDisposition
     
```
Added

jannahastings commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -387,43 +376,62 @@ Class: <http://openenergy-platform.org/ontology/oeo-physical/PortionOfNitrousOxi
     
     
-Class: <http://openenergy-platform.org/ontology/oeo-physical/RenewableEnergyCarrier>
+Class: <http://openenergy-platform.org/ontology/oeo-physical/RenewableFuel>
```
Definition updated

jannahastings commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -387,43 +376,62 @@ Class: <http://openenergy-platform.org/ontology/oeo-physical/PortionOfNitrousOxi
+Class: <http://openenergy-platform.org/ontology/oeo-physical/WasteRole>
 
     Annotations: 
         <http://purl.obolibrary.org/obo/IAO_0000115> "Waste is a role of an object aggregate that has been discarded after primary use."
```
Changed the name in the def to WasteRole

jannahastings commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -542,6 +562,16 @@ Class: AmbientHeat
     
     
+Class: Anthracite
+
```
Changed the def, added a comment with the further explanation

jannahastings commented:
OK, I accepted the suggestions and tried to address all comments, please have another look?

l-emele commented:
Currently I am busy but I will do a review today late afternoon.

akleinau commented:
RenewableFuel isn't a subclass of CombustionFuel yet

jannahastings commented:
> RenewableFuel isn't a subclass of CombustionFuel yet

OK, I added the axiom RenewableFuel 'has disposition' some CombustibleEnergyCarrierDisposition and now it is in the inferred view

jannahastings commented:
@l-emele I added the missing axioms to generate the missing inferences you highlighted. Please take another look. 

One thing I wasn't sure of, once I removed SolidBiomass from SolidFossilFuel, Coal is inferred as a solid fossil fuel but Peat is not, as it doesn't have the axiom that it has origin fossil. Is that correct?

l-emele commented:
Quick response: Peat should also have a fossil origin and hence be inferred as SolidFossilFuel.

I will do a full review later today.

jannahastings commented:
> Peat should also have a fossil origin

OK -- now added!

akleinau commented:
I'll fix the conflicts

akleinau commented:
fixed. I also deleted the PortionOf prefix of FluorinatedGreenhouseGas

l-emele commented:
Additional question: Is there a reason why `SolidBiomass` is treated differently to `LiquidBiofuel`?
![grafik](https://user-images.githubusercontent.com/36884905/76979125-d5d75c00-6937-11ea-9c91-02d2b15f5f42.png)

jannahastings commented:
I added the missing axioms to make SolidBiomass inferred as a Biofuel and Peat as a SolidBiomass. Note that "peat" now has two origins: biogenic and fossil. So it is inferred as both a fossil and a biofuel. @l-emele is that right?

jannahastings commented:
I also deleted the 'oil' class

l-emele commented:
> 
> 
> I added the missing axioms to make SolidBiomass inferred as a Biofuel and Peat as a SolidBiomass. Note that "peat" now has two origins: biogenic and fossil. So it is inferred as both a fossil and a biofuel. @l-emele is that right?

Hm, that's a difficult one. I just look up what the IPCC in their [FAQ](https://www.ipcc-nggip.iges.or.jp/faq/faq.html):
> Q2-7.	Is peat classified as fossil fuel or as biofuel in the _IPCC Guidelines_? What is the scientific basis for the classification the IPCC has chosen?
> A. In terms of the _Revised 1996 Guidelines_, peat or peatland appear in the Energy and the Land-Use Change and Forestry (LUCF) chapters. Peat is classified as "Solid Fossil" in the Energy Sector.
> In the _2006 Guidelines_ peat is not described as a fossil fuel or as biomass but as peat because “Although peat is not strictly speaking a fossil fuel, its greenhouse gas emission characteristics have been shown in life cycle studies to be comparable to that of fossil fuels (Nilsson and Nilsson, 2004; Uppenberg et al., 2001; Savolainen et al., 1994). Therefore, the CO2 emissions from combustion of peat are included in the national emissions as for fossil fuels.”  (See the footnote 5 on page 1.15 in Chapter 1, Volume 2 in the 2006 Guidelines.)

[Wikipedia](https://en.wikipedia.org/w/index.php?title=Peat&oldid=943599923) sums this up to:
> The Intergovernmental Panel on Climate Change (IPCC) classifies peat as neither a fossil fuel nor a renewable fuel, and notes that its emission characteristics are similar to fossil fuels.

For energy `modelling` (our scope) one usually treats peat simply as fossil origin. So I suggest that we use in our ontology only the fossil origin.

@stap-m @han-f @Ludee : What do you think?

l-emele commented:
I am very sorry. I mixed something up. :anguished:
In my comment above I wrote:
> The `SolidBiomass` still does appear correctly in inferred view. Only `WoodAndOther` is currently a `Biofuel`. Also `Peat` should be a `SolidBiomass`. So in the end the hierarchy should be:
> * `Biofuel`      
>   * `SolidBiomass`
>       * `Peat`
>       * `WoodAndOtherProducts`

But I confused there `Peat` and `CharCoal`. I should have written:
> The `SolidBiomass` still does appear correctly in inferred view. Only `WoodAndOther` is currently a `Biofuel`. Also **`CharCoal`** should be a `SolidBiomass`. So in the end the hierarchy should be:
> * `Biofuel`      
>   * `SolidBiomass`
>       * **`Charcoal`**
>       * `WoodAndOtherProducts`

This explains why `Peat` got accidentally both fossil and biogenic origins.

l-emele commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -291,6 +313,18 @@ pull request: https://github.com/OpenEnergyPlatform/ontology/pull/276"
+Class: <http://openenergy-platform.org/ontology/oeo-physical/FluorinatedGreenhouseGas>
+
+    Annotations: 
+        <http://purl.obolibrary.org/obo/IAO_0000115> "FluorinatedGreenhouseGas is a GreenhouseGas that is produced by fluorination. Hence they contains Fluor (F) atoms and are of anthropogenic origin."
```
```suggestion
        <http://purl.obolibrary.org/obo/IAO_0000115> "A FluorinatedGreenhouseGas is a GreenhouseGas that is produced by fluorination. Hence it contains Fluor (F) atoms and is of anthropogenic origin."
```

l-emele commented on src/ontology/edits/oeo-physical.omn:
```suggestion
        <http://purl.obolibrary.org/obo/IAO_0000115> "Motor spirit prepared especially for aviation piston engines, with an octane number suited to the engine, a freezing point of -60 °C and a distillation range usually within the limits of 30 °C and 180 °C."@en,
```

l-emele commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -1906,628 +2378,115 @@ pull request: https://github.com/OpenEnergyPlatform/ontology/pull/234"
     
     
-Class: PortionOfAir
+Class: PortionOfJetFuel
```
```suggestion
Class: JetFuel
```
The prefix PortionOf not needed anymore.

jannahastings commented:
Okay :-)  @l-emele I removed the biogenic origin of peat and added it for charcoal. Could you have another look?

stap-m commented:
> For energy modelling (our scope) one usually treats peat simply as fossil origin. So I suggest that we use in our ontology only the fossil origin.

Thanks @l-emele, I agree. Our 'fossil' definition tolerably fits for peat, too: *x is fossil if it was created from organic material by geolocial processes lasting thousands or millions of years.*

l-emele commented:
Yay! Thank you for all your effort solving this very large and difficult issue!
